terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -28,10 +28,26 @@ class FileParser < Dependabot::FileParsers::Base
       def parse
         dependency_set = DependencySet.new
 
+        parse_terraform_files(dependency_set)
+
+        parse_terragrunt_files(dependency_set)
+
+        dependency_set.dependencies.sort_by(&:name)
+      end
+
+      private
+
+      def parse_terraform_files(dependency_set)
         terraform_files.each do |file|
           modules = parsed_file(file).fetch("module", {})
           modules.each do |name, details|
-            dependency_set << build_terraform_dependency(file, name, details)
+            details = details.first
+
+            source = source_from(details)
+            # Cannot update local path modules, skip
+            next if source[:type] == "path"
```

Comment:
Local source paths can't have versions as far as I know, not sure how they would be specified.

Reply:
I couldn't find anything confirming or denying whether it is possible to add versions in local source paths. The closest thing I could find is using the `depends_on` argument

> Use the depends_on meta-argument to handle hidden resource or module dependencies that Terraform can't automatically infer.
 Explicitly specifying a dependency is only necessary when a resource or module relies on some other resource's behavior but doesn't access any of that resource's data in its arguments.

Because the parent module wouldn't access have any of its child module's data here, including the version, I think this is a safe enough optimization to include with this PR.